modules/sdk-lib-mpc/test/unit/tss/ecdsa/dklsDkg.ts

```diff
@@ -0,0 +1,241 @@
+  it(`should create key shares`, async function () {
+    const user = new DklsDkg.Dkg(3, 2, 0);
+    const backup = new DklsDkg.Dkg(3, 2, 1);
+    const bitgo = new DklsDkg.Dkg(3, 2, 2);
+    const userRound1Message = await user.initDkg();
+    const backupRound1Message = await backup.initDkg();
+    const bitgoRound1Message = await bitgo.initDkg();
+    const userRound2Messages = user.handleIncomingMessages({
+      p2pMessages: [],
+      broadcastMessages: [bitgoRound1Message, backupRound1Message],
+    });
+    const backupRound2Messages = backup.handleIncomingMessages({
+      p2pMessages: [],
+      broadcastMessages: [userRound1Message, bitgoRound1Message],
+    });
+    const bitgoRound2Messages = bitgo.handleIncomingMessages({
+      p2pMessages: [],
+      broadcastMessages: [userRound1Message, backupRound1Message],
+    });
+    const userRound3Messages = user.handleIncomingMessages({
+      p2pMessages: backupRound2Messages.p2pMessages
+        .filter((m) => m.to == 0)
+        .concat(bitgoRound2Messages.p2pMessages.filter((m) => m.to == 0)),
+      broadcastMessages: [],
+    });
+    const backupRound3Messages = backup.handleIncomingMessages({
+      p2pMessages: bitgoRound2Messages.p2pMessages
+        .filter((m) => m.to == 1)
+        .concat(userRound2Messages.p2pMessages.filter((m) => m.to == 1)),
+      broadcastMessages: [],
+    });
+    const bitgoRound3Messages = bitgo.handleIncomingMessages({
+      p2pMessages: backupRound2Messages.p2pMessages
+        .filter((m) => m.to == 2)
+        .concat(userRound2Messages.p2pMessages.filter((m) => m.to == 2)),
+      broadcastMessages: [],
+    });
+    const userRound4Messages = user.handleIncomingMessages({
+      p2pMessages: backupRound3Messages.p2pMessages
+        .filter((m) => m.to == 0)
+        .concat(bitgoRound3Messages.p2pMessages.filter((m) => m.to == 0)),
+      broadcastMessages: [],
+    });
+    const backupRound4Messages = backup.handleIncomingMessages({
+      p2pMessages: bitgoRound3Messages.p2pMessages
+        .filter((m) => m.to == 1)
+        .concat(userRound3Messages.p2pMessages.filter((m) => m.to == 1)),
+      broadcastMessages: [],
+    });
+    const bitgoRound4Messages = bitgo.handleIncomingMessages({
+      p2pMessages: backupRound3Messages.p2pMessages
+        .filter((m) => m.to == 2)
+        .concat(userRound3Messages.p2pMessages.filter((m) => m.to == 2)),
+      broadcastMessages: [],
+    });
+    user.handleIncomingMessages({
+      p2pMessages: [],
+      broadcastMessages: bitgoRound4Messages.broadcastMessages.concat(backupRound4Messages.broadcastMessages),
+    });
+    bitgo.handleIncomingMessages({
+      p2pMessages: [],
+      broadcastMessages: backupRound4Messages.broadcastMessages.concat(userRound4Messages.broadcastMessages),
+    });
+    backup.handleIncomingMessages({
+      p2pMessages: [],
+      broadcastMessages: bitgoRound4Messages.broadcastMessages.concat(userRound4Messages.broadcastMessages),
+    });
```

Comment:
maybe add a check that all three parties get to the same public key in their keyshares.

Reply:
done